test/extensions/network/dns_resolver/cares/dns_impl_test.cc

```diff
@@ -477,7 +478,12 @@ TEST_F(DnsImplConstructor, SupportsCustomResolversAsFallback) {
   int result = ares_get_servers_ports(peer->channel(), &resolvers);
   EXPECT_EQ(result, ARES_SUCCESS);
   EXPECT_EQ(resolvers->family, AF_INET);
-  EXPECT_STRNE(inet_ntop(AF_INET, &resolvers->addr.addr4, addr4str, INET_ADDRSTRLEN), "1.2.3.4");
+  if (peer->isCaresDefaultTheOnlyNameserver()) {
```

Comment:
Sorry I'm confused. How does this fail for you and not anyone else? What is different about your environment?

Reply:
Is it that you don't have a working resolve.conf per the above comment? I think I probably just don't have enough context on what this test is testing.

Reply:
yeah no working resolve.conf, so only the default resolver, at which point the resolver will always use failovers (even if they're configured to only be failovers)
the test tests that failovers aren't used, and so fails for me.

Reply:
`isCaresDefaultTheOnlyNameserver` gets called from `initializeChannel` which gets called multiple times throughout the session. Are we sure this doesn't change existing behavior? Either way do you mind adding some more comments in this test to say why we branch on this (what the host environment would need to look at)? I'm worried someone else is going to come along and get confused?

/wait